src/snowcli/exception.py

```diff
@@ -18,3 +18,8 @@ def format_message(self):
 class SnowflakeConnectionError(ClickException):
     def __init__(self, snowflake_err: Exception):
         super().__init__(f"Could not connect to Snowflake. Reason: {snowflake_err}")
+
+
+class UnsupportedConfigSectionType(ClickException):
```

Comment:
IMO we should extend ClickException only for expected exceptions - like connection error, invalid configuration etc.

This new exception can be thrown only in case of a bug (in our or 3rd party plugin's code) and I'm not sure if we should hide traceback for it in `--debug` mode (ClickExceptions are handled by click and they are not propagated further, so we can't get traceback).